### PR TITLE
[frontend] 1205 - Delete an item from a list (inject, atomic testing,…

### PR DESCRIPTION
**Pull request by Christian-DONGMO**
… scenario, simlation)

### Proposed changes

*
*

### Related issues

*
*

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: openbas-front/src/components/common/pagination/PaginationComponent.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ const useStyles = makeStyles(() => ({
 const ROWS_PER_PAGE_OPTIONS = [20, 50, 100];
 
 interface Props<T> {
+  key?: React.Key;
```
Comment (RomuDeuxfois) [inline openbas-front/src/components/common/pagination/PaginationComponent.tsx:44]: This new param seems to not be used

File: openbas-front/src/components/common/pagination/PaginationComponent.tsx [hunk 2]
```diff
@@ -101,7 +102,7 @@ const PaginationComponent = <T extends object>({
   // Filters
   const [openMitreFilter, setOpenMitreFilter] = React.useState(false);
 
-  useEffect(() => {
+  const fetchData = () => {
```
Comment (RomuDeuxfois) [inline openbas-front/src/components/common/pagination/PaginationComponent.tsx:105]: Why this change ?

If fetchData is not called by any other component.

File: openbas-api/src/main/java/io/openbas/utils/pagination/SearchPaginationInput.java [hunk 3]
```diff
@@ -22,7 +22,7 @@ public class SearchPaginationInput {
   @Min(0) int page = 0;
 
   @Schema(description = "Element number by page")
-  @Max(1000) int size = 10;
+  @Max(1000) int size = 20;
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/utils/pagination/SearchPaginationInput.java:25]: Why this change ?

File: openbas-front/src/admin/components/Dashboard.tsx [hunk 4]
```diff
@@ -222,6 +226,7 @@ const Dashboard = () => {
             exercises={exercises}
             searchPaginationInput={searchPaginationInput}
             setSearchPaginationInput={setSearchPaginationInput}
+            onOperationSuccess={refreshExercises}
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/Dashboard.tsx:229]: Can you explain why we need this new mechanism ?
I don't understand

File: openbas-front/src/admin/components/scenarios/Scenarios.tsx [hunk 5]
```diff
@@ -194,22 +194,14 @@ const Scenarios = () => {
...
-  const handleOpenExport = (scenarioId: string) => {
-    setOpenExportId(scenarioId);
-  };
-  const handleCloseExport = () => {
-    setOpenExportId(null);
+  const refreshScenarios = () => {
+    searchScenarios(searchPaginationInput).then((result) => {
+      const { data } = result;
+      setScenarios(data.content);
+      setModificationTrigger((prev) => !prev);
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/scenarios/Scenarios.tsx:203]: I don't understand the usage of modificationTrigger

File: openbas-front/src/components/common/pagination/PaginationComponent.tsx (hunk 1 above)
Comment (Christian-DONGMO) [reply openbas-front/src/components/common/pagination/PaginationComponent.tsx:44]: This param is used in Scenarios.tsx to force the pagination to refresh in the case of a duplication or deletion of an element from the list

File: openbas-front/src/components/common/pagination/PaginationComponent.tsx (hunk 2 above)
Comment (Christian-DONGMO) [reply openbas-front/src/components/common/pagination/PaginationComponent.tsx]: Forgetting to go back during a test

File: openbas-api/src/main/java/io/openbas/utils/pagination/SearchPaginationInput.java (hunk 3 above)
Comment (Christian-DONGMO) [reply openbas-api/src/main/java/io/openbas/utils/pagination/SearchPaginationInput.java:25]: I found that when I modify an element in the list, and I propagate the change to update the list, it is reloaded but uses the default value set in the backend. And there is an inconsistency between the minimum value of the back which is 10 and that of the front which is 20

File: openbas-front/src/admin/components/Dashboard.tsx (hunk 4 above)
Comment (Christian-DONGMO) [reply openbas-front/src/admin/components/Dashboard.tsx:229]: When you add or delete a simulation from the Dashboard, the list is not updated. This new mechanism allows me to propagate the change to the parent and forces it to update the list

File: openbas-front/src/admin/components/scenarios/Scenarios.tsx (hunk 5 above)
Comment (Christian-DONGMO) [reply openbas-front/src/admin/components/scenarios/Scenarios.tsx:203]: modificationTrigger just allows me to signal a change of state which I transmit to the PaginationComponent component

File: openbas-front/src/components/common/pagination/PaginationComponent.tsx (hunk 1 above)
Comment (RomuDeuxfois) [reply openbas-front/src/components/common/pagination/PaginationComponent.tsx:44]: Ok but the parameter is not used in the class. It looks like magic.

And by making this change, we bypass the use of the store.

What is the use case?

File: openbas-api/src/main/java/io/openbas/utils/pagination/SearchPaginationInput.java (hunk 3 above)
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/utils/pagination/SearchPaginationInput.java:25]: well seen

File: openbas-front/src/admin/components/Dashboard.tsx (hunk 4 above)
Comment (RomuDeuxfois) [reply openbas-front/src/admin/components/Dashboard.tsx:229]: Can you show me where it's possible to add or delete a simulation from the Dashboard screen ?

**Review by RomuDeuxfois — CHANGES_REQUESTED** [review]

File: openbas-front/src/admin/components/simulations/simulation/ExercisePopover.tsx [hunk 6]
```diff
@@ -152,7 +160,7 @@ const ExercisePopover: FunctionComponent<ExercisePopoverProps> = ({
   // Button Popover
   const entries = [];
   if (actions.includes('Update')) entries.push({ label: 'Update', action: () => handleOpenEdit(), disabled: !permissions.canWriteBypassStatus });
-  if (actions.includes('Delete')) entries.push({ label: 'Delete', action: () => handleOpenDelete(), disabled: !permissions.canWriteBypassStatus });
```
[the remark this answers is not in the document]
Comment (RomuDeuxfois) [reply openbas-front/src/admin/components/simulations/simulation/ExercisePopover.tsx:155]: We need to talk with the product team about this.
